### PR TITLE
Run tests in cwd

### LINKED ISSUE #984: `spago test` runs script in package's directory, not workspace directory
See https://github.com/JordanMartinez/spago-next-bug for a reproducible example, which surprised me.

I think there are three responses to this issue.:
1. always run `spago test` in the workspace directory
1. always run `spago test` in the package directory
1. always run `spago test` in the directory in the current working directory

Most of the time, if I'm running a test, I execute the `spago test` command when my current working directory is the workspace directory.

### PR DESCRIPTION
**Pull request by JordanMartinez**
### Description of the change

Fixes #984.

### Checklist:

- [ ] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [x] Added a test for the contribution (if applicable)

**P.S.**: the above checks are not compulsory to get a change merged, so you may skip them. However, taking care of them will result in less work for the maintainers and will be much appreciated 😊

### CODE REVIEW

**Comment by JordanMartinez** [comment]
There are tests unrelated to this PR that currently fail locally. So, the third commit above proves that this change works. Once proven by CI, I'll revert the third commit.

**Comment by f-f** [comment]
@JordanMartinez the CI for the PureScript side of things is not running yet - I am fixing it at the moment, so that we'll be able to get meaningful reporting here on GH

**Comment by JordanMartinez** [comment]
Oh! That would make sense.

**Comment by f-f** [comment]
We have PureScript CI now! Could you rebase on the latest commit? I had to shuffle files around unfortunately, so it might be easier to just copy the patch to a new branch instead of trying to merge.

**Comment by JordanMartinez** [comment]
Sure. I'll get to that when I can.

File: test/Prelude.purs [hunk 1]
```diff
@@ -66,8 +68,15 @@ checkFixture filepath fixturePath = do
...
-checkResultAndOutputs maybeOutFixture maybeErrFixture resultFn execResult = do
+plusDependencies :: Array String -> Config -> Config
+plusDependencies deps config = config
+  { package = config.package <#> \p -> p { dependencies = p.dependencies <> (Dependencies $ Map.fromFoldable $ map mkDep deps) }
+  }
+  where
+  mkDep p = Tuple (unsafeFromRight $ PackageName.parse p) Nothing
+
+checkResultAndOutputs' :: Maybe String -> Maybe FilePath -> (Either ExecError ExecResult -> Boolean) -> Either ExecError ExecResult -> Aff _
+checkResultAndOutputs' maybeOutStr maybeErrStr resultFn execResult = do
```
Comment (JordanMartinez) [inline test/Prelude.purs:79]: Thoughts on renaming this to something else? The `'` isn't much of a difference and the type signature is the same as `checkResultAndOutputs`. So, I see this as an easy gotcha when writing tests in the future.

Comment (f-f) [reply test/Prelude.purs:79]: Good point. How's `checkResultAndOutputsStr`?

Comment (JordanMartinez) [reply test/Prelude.purs]: Done.

**Comment by f-f** [comment]
What if you write a file in the root of the repo and try to read its content from the script in the test? Then we don't need to verify what the cwd is, just that we were able to read the file from the right place.

**Comment by f-f** [comment]
It looks like some macOS tests are just timing out, let's try to bump the general test timeout from 30s to maybe 60? (It's in `test/Spago.purs`)

File: test-fixtures/spago-subpackage-test-cwd.purs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import Node.Encoding (Encoding(..))
+import Node.FS.Aff (readTextFile)
+
+main :: Effect Unit
+main = launchAff_ do
+  readTextFile UTF8 filePath >>= log
+
+-- DO NOT EDIT. This value is replaced in the corresponding spago test
+filePath :: String
+filePath = "FILE_PATH"
```
Comment (f-f) [inline test-fixtures/spago-subpackage-test-cwd.purs:17]:
```suggestion
filePath = "./foo.txt"
```

I think we don't need to do the whole replacing thing, but can just use a relative path?

File: test/Prelude.purs [hunk 3]
```diff
@@ -66,8 +68,15 @@ checkFixture filepath fixturePath = do
...
-checkResultAndOutputs :: Maybe FilePath -> Maybe FilePath -> (Either ExecError ExecResult -> Boolean) -> Either ExecError ExecResult -> Aff _
-checkResultAndOutputs maybeOutFixture maybeErrFixture resultFn execResult = do
+plusDependencies :: Array String -> Config -> Config
+plusDependencies deps config = config
+  { package = config.package <#> \p -> p { dependencies = p.dependencies <> (Dependencies $ Map.fromFoldable $ map mkDep deps) }
+  }
+  where
+  mkDep p = Tuple (unsafeFromRight $ PackageName.parse p) Nothing
+
+checkResultAndOutputsStr :: Maybe String -> Maybe FilePath -> (Either ExecError ExecResult -> Boolean) -> Either ExecError ExecResult -> Aff _
```
Comment (f-f) [inline test/Prelude.purs:78]:
```suggestion
checkResultAndOutputsStr :: Maybe String -> Maybe String -> (Either ExecError ExecResult -> Boolean) -> Either ExecError ExecResult -> Aff _
```

File: test-fixtures/spago-subpackage-test-cwd.purs (hunk 2 above)
Comment (thomashoneyman) [reply test-fixtures/spago-subpackage-test-cwd.purs:17]: Looks like spago has historically had windows CI, so should this be (“.” <> Path.sep <> “file.txt”) instead?

Comment (f-f) [reply test-fixtures/spago-subpackage-test-cwd.purs:17]: It might also work to just use `foo.txt`

**Comment by JordanMartinez** [comment]
> It looks like some macOS tests are just timing out, let's try to bump the general test timeout from 30s to maybe 60? (It's in `test/Spago.purs`)

I'm pretty sure it's node-execa that's at fault. In the latest release where I updated it to the new node libraries, I came across this issue thanks to Verity: https://github.com/JordanMartinez/purescript-node-execa/commit/fda206466d694fd3bb042b0f6cb758a0aea9adae

That version of the dependency used here might still have that code.

**Comment by f-f** [comment]
> I'm pretty sure it's node-execa that's at fault

I would try to extend the timeout first - tests are passing on my mac.

If it turns out it's not that then we can consider upgrading to the latest set (probably in a separate PR)

**Comment by JordanMartinez** [comment]
> > I'm pretty sure it's node-execa that's at fault
> 
> I would try to extend the timeout first - tests are passing on my mac.
> 
> If it turns out it's not that then we can consider upgrading to the latest set (probably in a separate PR)

Oh right. Tests pass on Linux, so it can't be that...

**Review by f-f — APPROVED** [review]
Great, thanks!

**Pull request merged**